etcdv3 backend acceptance tests fail due to locking issue

### Test Failure Output
```
--- FAIL: TestBackend (13.33s)
    backend_test.go:40: Cleaned up 0 keys.
    backend_test.go:62: TestBackendConfig on *etcd.Backend with configs.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"endpoints":cty.TupleVal([]cty.Value{cty.StringVal("http://127.0.0.1:2379")}), "prefix":cty.StringVal("tf-unit/2021-03-15T16:43:15-05:00/")}}
    backend_test.go:67: TestBackendConfig on *etcd.Backend with configs.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"endpoints":cty.TupleVal([]cty.Value{cty.StringVal("http://127.0.0.1:2379")}), "prefix":cty.StringVal("tf-unit/2021-03-15T16:43:15-05:00/")}}
    testing.go:299: TestBackend: testing state locking for *etcd.Backend
    testing.go:299: failed to read locked state from another backend instance: Failed to lock state in etcd: context deadline exceeded
        Lock Info:
          ID:        6ca7cb68-6fb0-a6e5-b53a-9c0aeea34771
          Path:      tf-unit/2021-03-15T16:43:15-05:00/default/5c4078370538cad8
          Operation: test
          Who:       clientA
          Version:   0.15.0
          Created:   2021-03-15 21:43:17.369859 +0000 UTC
          Info:      
        .
    testing.go:299: TestBackend: testing state locking for *etcd.Backend
    testing.go:299: failed to read locked state from another backend instance: Failed to lock state in etcd: context deadline exceeded
        Lock Info:
          ID:        3043508f-c881-c9fe-4d7e-b3b61d8254c8
          Path:      tf-unit/2021-03-15T16:43:15-05:00/default/5c4078370538cb04
          Operation: test
          Who:       clientA
          Version:   0.15.0
          Created:   2021-03-15 21:43:22.137153 +0000 UTC
          Info:      
        .
    backend_test.go:40: Cleaned up 4 keys.
FAIL
FAIL    github.com/hashicorp/terraform/backend/remote-state/etcdv3      17.263s
FAIL
```

### Expected Behavior
Acceptance tests should pass when run against a fresh etcdv3 instance.

### Actual Behavior
Acceptance tests fail due to a timeout when trying to read locked state from a second backend instance when the backend is already locked.

### Steps to Reproduce
1. Apply changes in https://github.com/hashicorp/terraform/pull/28097 and https://github.com/hashicorp/terraform/pull/28099
2. `helm install etcd bitnami/etcd --set auth.rbac.enabled=false` (installed in Minikube)
3. `kubectl port-forward --namespace default svc/etcd 2379:2379`
4. `TF_ACC=1 TF_ETCDV3_ENDPOINTS=http://127.0.0.1:2379 go test ./backend/remote-state/etcdv3`

### Additional Context
I only saw these failures after resolving two other issues with the acceptance tests listed below in the references.

### References
type conversion issue:
https://github.com/hashicorp/terraform/issues/28096
https://github.com/hashicorp/terraform/pull/28097

duplicate workspaces issue:
https://github.com/hashicorp/terraform/issues/28098
https://github.com/hashicorp/terraform/pull/28099

--- a/internal/backend/remote-state/etcdv3/backend_state.go
+++ b/internal/backend/remote-state/etcdv3/backend_state.go
@@ -58,16 +58,8 @@ func (b *Backend) StateMgr(name string) (statemgr.Full, error) {
 
 	lockInfo := statemgr.NewLockInfo()
 	lockInfo.Operation = "init"
-	lockId, err := stateMgr.Lock(lockInfo)
-	if err != nil {
-		return nil, fmt.Errorf("Failed to lock state in etcd: %s.", err)
-	}
-
 	lockUnlock := func(parent error) error {
-		if err := stateMgr.Unlock(lockId); err != nil {
-			return fmt.Errorf(strings.TrimSpace(errStateUnlock), lockId, err)
-		}
-		return parent
+		return nil
 	}
 
 	if err := stateMgr.RefreshState(); err != nil {
@@ -76,6 +68,18 @@ func (b *Backend) StateMgr(name string) (statemgr.Full, error) {
 	}
 
 	if v := stateMgr.State(); v == nil {
+		lockId, err := stateMgr.Lock(lockInfo)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to lock state in etcd: %s.", err)
+		}
+
+		lockUnlock = func(parent error) error {
+			if err := stateMgr.Unlock(lockId); err != nil {
+				return fmt.Errorf(strings.TrimSpace(errStateUnlock), lockId, err)
+			}
+			return parent
+		}
+
 		if err := stateMgr.WriteState(states.NewState()); err != nil {
 			err = lockUnlock(err)
 			return nil, err

